File: backend/src/app/models/gpon/GponCount.ts
import { BuildOptions, DataTypes, Model, Sequelize } from "sequelize";
import { GponIntegration } from "../../../_lib/database/main";

export interface GponCountAttributes {
    id: number;
    erp_ap_id: number;
    gpon_integration_id: number;
    description: string;
    online: number;
    offline: number;
}
export interface GponCountModel extends Model<GponCountAttributes>, GponCountAttributes { }

export type GponCountStatic = typeof Model & {
    new(values?: object, options?: BuildOptions): GponCountModel;
};

export function GponCountFactory(sequelize: Sequelize): GponCountStatic {
    return <GponCountStatic>sequelize.define("gpon_count", {
        id: {
            type: DataTypes.INTEGER,
            autoIncrement: true,
            primaryKey: true,
        },
        erp_ap_id: {
            type: DataTypes.INTEGER,
        },
        gpon_integration_id: {
            type: DataTypes.INTEGER,
            references: {
                model: GponIntegration,
                key: 'id'
            }
        },
        description: {
            type: DataTypes.STRING,
            allowNull: true,
        },
        online: {
            type: DataTypes.NUMBER,
        },
        offline: {
            type: DataTypes.NUMBER,
        },
    }, { tableName: "gpon_count", timestamps: false });
}

File: snmpagent/app/models/gpon/GponCount.py
from django.db import models
from app.models.Ap import Ap

class GponCount(models.Model):
    description = models.CharField(max_length=100)
    online = models.IntegerField(null=False)
    offline = models.IntegerField(null=False)

    erp_ap_id = models.IntegerField()
    gpon_integration_id = models.IntegerField()

    class Meta:
        db_table = "gpon_count"
    
    def __str__(self):
        return '{}|{}'.format(self.description, self.online)